object/token_jwt.go

```diff
@@ -288,6 +309,55 @@ func generateJwtToken(application *Application, user *User, nonce string, scope
 	return tokenString, refreshTokenString, name, err
 }
 
+func generateSubscriptionToken(
+	application *Application, user *User, subscription *Subscription, nonce string, scope string, host string,
+) (string, error) {
+	nowTime := time.Now()
+	expireTime := nowTime.Add(time.Duration(subscription.Duration) * time.Hour * 24)
+
+	user = refineUser(user)
+
+	_, originBackend := getOriginFromHost(host)
+
+	name := util.GenerateId()
+	jti := fmt.Sprintf("%s/%s", application.Owner, name)
+
+	claims := Claims{
+		User:      user,
+		TokenType: "subscription-token",
```

Comment:
generateSubscriptionToken mostly the same as generateSubscriptionToken
        I think possible to deduplicate code by pass tokenType as arg 

	
	if tokenType != "subscription-token" {
		// the JWT token length in "JWT-Empty" mode will be very short, as User object only has two properties: owner and name
		if application.TokenFormat == "JWT-Empty" {
			claimsShort := getShortClaims(claims)

			token = jwt.NewWithClaims(jwt.SigningMethodRS256, claimsShort)
			claimsShort.ExpiresAt = jwt.NewNumericDate(refreshExpireTime)
			claimsShort.TokenType = "refresh-token"
			refreshToken = jwt.NewWithClaims(jwt.SigningMethodRS256, claimsShort)
		} else {
			claimsWithoutThirdIdp := getClaimsWithoutThirdIdp(claims)

			token = jwt.NewWithClaims(jwt.SigningMethodRS256, claimsWithoutThirdIdp)
			claimsWithoutThirdIdp.ExpiresAt = jwt.NewNumericDate(refreshExpireTime)
			claimsWithoutThirdIdp.TokenType = "refresh-token"
			refreshToken = jwt.NewWithClaims(jwt.SigningMethodRS256, claimsWithoutThirdIdp)
		}
	}